scripts/superdesk-search/search.js

```diff
@@ -2050,13 +2070,14 @@
                             var itemsById = angular.extend({}, listComponent.state.itemsById);
 
                             items._items.forEach(function(item) {
-                                if (!itemsById[item._id] || !isSameVersion(itemsById[item._id], item)) {
-                                    itemsById[item._id] = item;
+                                var itemId = search.generateTrackByIdentifier(item);
```

Comment:
We get `_id` of the document in the item:lock and item:unlock notification. 
https://github.com/akintolga/superdesk-client-core/blob/cd-4093/scripts/superdesk-search/search.js#L2098
we won't be able to find the item in the `itemsById[itemId]` to update the state.